### DISCUSSION TITLE
The export post-processing filter failed to render the file

### DISCUSSION THREAD

**Discussion by Silversinker** [ImageMagick/ImageMagick#7538] [Help]
### ImageMagick version

7.1.1-Q16

### Operating system, version and so on

Win 11

### Description

Hi,

I have had to reinstall windows and all apps after a significant hardware failure.
I have had LRC for several years and never had this problem before.

I am trying to export 2 files from LRC 13.4 with LR Mogrify text anotations:

CC25-07-24 (15).ARW
CC25-07-24 (12).jpg

Export as TIF or JPG no issues, export as PNG the error below occurs. The export settings are identical for each other than a change of extension.

**The export post-processing filter failed to render the file. (2)
D:\Pictures\Nature\CC\CC25-07-24 (15).ARW
D:\Pictures\Nature\CC\CC25-07-24 (12).jpg**

Why? Can anyone throw any light on this please.

**Comment by snibgo** [comment]
What is LRC?

**Reply by Silversinker** [reply]
Lightroom Classic, the desktop based version

**Comment by snibgo** [comment]
Your filenames seem to contain spaces. I suggest not doing this. Spaces in filenames cause complications.

Lightroom is an Adobe product.

If Lightroom uses ImageMagick, and uses IM as an installed system (rather than being built-in to Lightroom), then you should check that IM can write PNG and other files. Try these:
```
magick rose: r.png

magick rose: r.jpg

magick rose: r.tiff
```
If one of those fails, then try:
```
magick -list format
```
... and examine the entry for the format that failed, such as PNG.

**Reply by Silversinker** [reply]
Hi snibgo,

The filename makes no difference.  If I change it to Test, I get the same results.

IM is a plugin for Lightroom

**Comment by fmw42** [comment]
What version of Imagemagick is required by Lightroom?  Is it IM 6 or IM 7?  You now suggest that you have IM 7? Is Lightroom compatible with IM 7?

**Reply by Silversinker** [reply]
Hi fmw43,
It is version 7, the same as it was before the issues occured following the reinstall of windows.  It used to work fine

**Comment by fmw42** [comment]
According to Imagemagick formats page at https://imagemagick.org/script/formats.php#supported, ARW is read only as indicated by R (rather than RW for read and write). Imagemagick does not write to ARW.

"ARW | R | Sony Digital Camera Alpha Raw Image Format | Set -define dng:use-camera-wb=true to use the RAW-embedded color profile for Sony cameras. You can also set these options: use-auto-wb, no-auto-bright, and output-color."

**Reply by Silversinker** [reply]
Hi fmw42,
ARW is the original RAW file that is being proceessed, not the output file type.
IM/LR Mogrify will output to either JPG or TIF without any problem, but as soon as I set the file type to PNG the error occurs.

Prior to having to reinstall windows all worked fine & I could oputput to any file type.

**Reply by fmw42** [reply]
Check to see if you have the PNG delegate installed. You can do that by 

`magick -list format`

and look for a line saying PNG. Then look at the end of the line to see the version.

Do you even have a PNG line?  If so what is the version?

**Reply by Silversinker** [reply]
Hi fmw42, 
Yes, there is a PNG line, it reads
 PNG* PNG       rw-   Portable Network Graphics (libpng 1.6.43)

**Comment by snibgo** [comment]
Upthread, I suggested some tests like `magick rose: r.png`.

**Reply by Silversinker** [reply]
Hi snibgo,
Sorry, I'm not familiar with magick, it is part of a plugin, LR Mogrify, in Lightroom.  I set parameters in LR Mogrify that get passed to Magick, as per screenshot.
As I have mentioned, all used to work great until I had to reinstall everything.  All other export formats work, just not PNG
![Magick](https://github.com/user-attachments/assets/3cdfee4c-27c1-4b74-8587-43001e30d1c8)
